ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/TracingOptionsManager.java

```diff
@@ -74,21 +74,29 @@ public Map<String, String> getTracingOptions(Map<String, String> storedOptions)
 		return defaults;
 	}
 
-	public Map<String, String> getTracingTemplateCopy() {
-		return new HashMap<>(getTracingTemplate());
+	public Map<String, String> getTracingTemplateCopy(IProgressMonitor monitor) {
+		return new HashMap<>(getTracingTemplate(monitor));
 	}
 
-	private synchronized Map<String, String> getTracingTemplate() {
-		if (template == null) {
-			Map<String, String> temp = new HashMap<>();
-			IPluginModelBase[] models = PluginRegistry.getAllModels();
-			Arrays.stream(models).map(TracingOptionsManager::getOptions).filter(Objects::nonNull).forEach(p -> {
+	private synchronized Map<String, String> getTracingTemplate(IProgressMonitor monitor) {
+		if (template != null) {
+			return template;
+		}
+
+		Map<String, String> temp = new HashMap<>();
+		IPluginModelBase[] models = PluginRegistry.getAllModels();
+		SubMonitor subMonitor = SubMonitor.convert(monitor, models.length);
+
+		for (IPluginModelBase model : models) {
+			Properties options = TracingOptionsManager.getOptions(model);
+			if (options != null) {
 				@SuppressWarnings({ "rawtypes", "unchecked" })
-				Map<String, String> entries = (Map) p;
+				Map<String, String> entries = (Map) options;
 				temp.putAll(entries); // All entries are of String/String
-			});
-			template = temp;
+			}
+			subMonitor.split(1);
```

Comment:
Please move `subMonitor.split(1)` to the beginning of the loop.
`SubMonitor.split()` cleans up an previously created other children and therefore 'completes/calls `done()` for them.
Consequently calling split at the end of the loop has the effect that the first progress reporting for the loop happens after the second model has been processed because only then a pending submonitor-child exists.
This means a slight delay of the first feedbeck.

Reply:
Done, thanks for the tip!